contracts/ronin/BridgeTracking.sol

```diff
@@ -144,41 +147,54 @@ contract BridgeTracking is HasBridgeContract, HasValidatorContract, Initializabl
     address _operator
   ) external override onlyBridgeContract skipOnUnstarted {
     uint256 _period = _validatorContract.currentPeriod();
-    _trySyncPeriodStats();
-    ReceiptStats storage _stats = _receiptStats[_kind][_requestId];
+    _trySyncBuffer();
+    ReceiptTrackingInfo storage _receiptInfo = _receiptTrackingInfo[_kind][_requestId];
 
-    // Stores the ones vote for the (deposit/mainchain withdrawal) request which is not approved yet
-    if (_stats.approvedPeriod == 0) {
-      _stats.voters.push(_operator);
+    // When the vote is not approved yet, the voters are saved in the receipt info, and not increase ballot metric.
+    // The ballot metric will be increased later in the {handleVoteApproved} method.
+    if (_receiptInfo.approvedPeriod == 0) {
+      _receiptInfo.voters.push(_operator);
       return;
     }
 
     _increaseBallot(_kind, _requestId, _operator, _period);
   }
 
   /**
-   * Increases the ballot for the operator at a period.
+   * @dev Increases the ballot for the operator at a period.
    */
   function _increaseBallot(
     VoteKind _kind,
     uint256 _requestId,
     address _operator,
-    uint256 _period
+    uint256 _currentPeriod
   ) internal {
-    ReceiptStats storage _receiptInfo = _receiptStats[_kind][_requestId];
+    ReceiptTrackingInfo storage _receiptInfo = _receiptTrackingInfo[_kind][_requestId];
     if (_receiptInfo.voted[_operator]) {
       return;
     }
 
     _receiptInfo.voted[_operator] = true;
 
-    // Only records within a period
-    if (_receiptInfo.approvedPeriod == _period) {
-      if (_temporaryStats.info.totalBallotsOf[_operator] == 0) {
-        _temporaryStats.info.voters.push(_operator);
+    uint256 _trackedPeriod = _receiptInfo.trackedPeriod;
+
+    // Do not increase ballot for receipt that is neither in the buffer, nor in the most current tracked period.
+    // If the receipt is not tracked in a period, increase metric in buffer.
+    if (_trackedPeriod == 0) {
+      if (_bufferMetric.data.totalBallotsOf[_operator] == 0) {
+        _bufferMetric.data.voters.push(_operator);
+      }
+      _bufferMetric.data.totalBallots++;
+      _bufferMetric.data.totalBallotsOf[_operator]++;
+    }
+    // If the receipt is tracked in the most current tracked period, increase metric in the period.
+    else if (_trackedPeriod == _currentPeriod) {
+      PeriodVotingMetric storage _metric = _periodMetric[_trackedPeriod];
+      if (_metric.totalBallotsOf[_operator] == 0) {
+        _metric.voters.push(_operator);
```

Comment:
The `_periodMetric[period].voters` is not synced with `_bufferMetric` in `_trySyncBuffer`, in case we don't want to save this information, we may remove this push here for consistency.